caustic/lenses/base.py

```diff
@@ -14,7 +14,13 @@
 
 __all__ = ("ThinLens", "ThickLens")
 
-class ThickLens(Parametrized):
+class Lens(Parametrized):
```

Comment:
Why add this class? It does nothing in the inheritance tree. I think we should add that only when we need it, for example if we have control flow common to ThickLens and ThinLenses

Reply:
There is actually some stuff I was planning to move into this class since it is shared by both ThinLens and ThickLens. Also, it is nice in general to be able to test if an object is a Lens object as opposed to say a Light object, Cosmology, etc.

But I can move this to a future PR if that's easier.